server/controllers/wishlistController.js

```diff
@@ -0,0 +1,202 @@
+const Wishlist = require('../models/wishlist');
+
+const _getWishlistCost = list => {
+  // Accept list as Array of objects
+  // Should return a single cost for the entire wishlist
+  let totalCost = 0;
+
+  list.forEach(item => {
+    totalCost += item.itemCost * item.itemQuantity;
+  });
+
+  return totalCost;
+};
+
+exports.addItem = (req, res) => {
+  // Bad request if no body is sent
+  if (Object.keys(req.body).length === 0) {
+    return res.status(400).json({
+      Error: 'Body missing data',
+    });
+  }
+
+  const familyId = req.params.familyId;
+  const item = req.body.item;
+
+  // TODO: Check structure of item
+  // Prevent duplicate items
+
+  try {
+    Wishlist.find({ family: familyId }).exec((err, wishlist) => {
```

Comment:
I'm partial to using the async/await and promise syntax instead of a callback function.  I think we should be consistent in this for the backend.  The only issue is that it is only supported in node >= 7. @rainrivas can you weigh in?

Reply:
I'm more comfortable with callbacks, but i do agree that we should be consistent. I don't mind going all async/await since LTS is now NODE 8 so 🤷‍♂️ on supporting the lower versions. Besides, can't we babel-magic it for those?